CONTRIBUTING.md

```diff
@@ -1,11 +1,83 @@
+* Please include the version of the Design Library you are using/viewing
+* When submitting a browser bug, please include the browser, version, operating system, and operating system version.
+* Include screenshots and animated GIFs whenever possible; they are immensely helpful.
+
+## Pull Requests
+
+* **DO NOT ISSUE A PULL REQUEST WITHOUT FIRST [SUBMITTING AN ISSUE](#submitting-issues)**
+* Pull requests should reference their related issues. If the pull request closes an issue, [please reference its closing in your commit messages](https://help.github.com/articles/closing-issues-via-commit-messages/). Pull requests not referencing any issues will be closed.
+* Update the `CHANGELOG` with the changes made by your pull request, making sure to use the proper [Emoji](#emoji-cheatsheet).
+* Make sure you have [installed the development environment](https://github.com/IBM-Watson/design-library#installation), [updated your runner to the latest version](https://github.com/IBM-Watson/design-library#updating-the-runner), and have [run the library locally](https://github.com/IBM-Watson/design-library#running-locally) to ensure that your code works properly.
+* Follow our JavaScript and CSS styleguides. We have linters set up to catch most of it.
+* Ensure that you have [EditorConfig](http://editorconfig.org/) installed in your editor of choice and that it is functioning properly.
+* Do not squash or rebase your commits when submitting a Pull Request. It makes it much harder to follow your work and make incremental changes.
+
+### Git Commit Messages
+
+* Use the present tense (`"Add feature"` not `"Added Feature"`)
+* Use the imperative mood (`"Move cursor to…"` not `"Moves cursor to…"`)
+* Limit the first line to 72 characters or less
+* Include relevant Emoji from our [Emoji cheatsheet](#emoji-cheatsheet)
+
+### Branching Model
+
+* Pull requests must be made into our [develop](https://github.com/IBM-Watson/design-library/tree/develop) branch.
+* The following branch prefixes should be used when creating a new branch:
+  * `hotfix/` - bug fixes that got through and need to be squashed
+  * `pattern/` - update to or new pattern to be added
+  * `language/` - update to or new piece of long-form text or assets to go along with it
+  * `library/` - update to or new code for the site
+  * `release/` - for releases
+  * `feature/` - update to or new general feature not covered by other prefixes
+
+## Creating a New Version
+
+Versioning is done through [SEMVER](http://semver.org/). When creating a new version, issue a [pull request](#pull-requests) from `develop` into `master` and create new release branch off of `master` with the version's name, and create a new tag with `v` prefixed with the version's name from that branch. 
+
+For instance, if you are creating version `1.1.0`, you would start by merging `develop` into `master`, create a branch `release/1.1.0` from `master`, and create a tag `v1.1.0` from branch `release/1.1.0`.
+
+## Emoji Cheatsheet
+
```

Comment:
I approve of the ones selected - I have the [Emoji Guide wiki page](https://github.com/IBM-Watson/design-library/wiki/Emoji-Guide) which has two additional emojis for issues I don't know if we want to include those as well.